opal/mca/btl/openib/btl_openib_component.c

```diff
@@ -622,6 +622,22 @@ static int init_one_port(opal_list_t *btl_list, mca_btl_openib_device_t *device,
     union ibv_gid gid;
     uint64_t subnet_id;
 
+/*
+ * Starting with Open MPI 4.0 we don't support infiniband
+ * unless the user specifically requested to override this
+ * policy.
+ */
+#if HAVE_DECL_IBV_LINK_LAYER_ETHERNET
```

Comment:
So this only does the filtering `IBV_LINK_LAYER_ETHERNET` exists -- e.g., with inbox verbs on RHEL 6.x, this test won't do anything because `IBV_LINK_LAYER_ETHERNET` doesn't exist.

Do we care?

I *kinda* care, because I think RHEL 6 still is not uncommon (it hasn't died yet).

It *might* be worthwhile to have the test try to do something sane on platforms such as RHEL 6 (e.g., exclude all IB transports -- even if it's RoCE -- with the rationale that the user can use the MCA param if they need to override that, yadda yadda yadda).  Would need explaining in the README and/or FAQ, though.

That's likely *better* (because it will *actually* disable IB on more platforms), but I don't know if the RM's want to go to that effort.  I don't have the cycles to help, sorry... 😢